scriptworker/test/test_config.py

```diff
@@ -74,6 +74,19 @@ def test_check_config_bad_keyring(t_config):
     assert "needs to start with %(gpg_home)s/" in "\n".join(messages)
 
 
+def test_check_config_invalid_ids(t_config):
+    t_config['provisioner_id'] = 'twenty-three-characters'
+    t_config['worker_group'] = 'twenty-three-characters'
+    t_config['worker_type'] = 'twenty-three-characters'
+    t_config['worker_id'] = 'twenty-three-characters'
+
+    messages = config.check_config(t_config, "test_path")
+    assert 'provisioner_id doesn\'t match "^[a-zA-Z0-9-_]{1,22}$" (required by Taskcluster)' in "\n".join(messages)
```

Comment:
I don't think you need to join the messages with '\n'; `member in list` works as long as member is a complete message.  Also, you join messages with '\n' multiple times.  I also don't think this is a huge issue.